crates/turborepo-api-client/src/lib.rs

```diff
@@ -213,6 +224,37 @@ impl APIClient {
         })
     }
 
+    pub async fn get_spaces(&self, token: &str, team_id: Option<&str>) -> Result<SpacesResponse> {
+        // create url with teamId if provided
+        let endpoint = match team_id {
+            Some(team_id) => format!("/v0/spaces?limit=100&teamId={}", team_id),
+            None => "/v0/spaces?limit=100".to_string(),
+        };
+
+        let response = self
+            .make_retryable_request(|| {
+                let request_builder = self
+                    .client
+                    .get(self.make_url(endpoint.as_str()))
+                    .header("User-Agent", self.user_agent.clone())
+                    .header("Content-Type", "application/json")
+                    .header("Authorization", format!("Bearer {}", token));
```

Comment:
Have we already required a `token` to be set or could it be `nil` here?

Reply:
We have already required it